lib/cartopy/io/img_tiles.py

```diff
@@ -350,23 +355,33 @@ class Stamen(GoogleWTS):
     """
 
     def __init__(self, style='toner',
-                 desired_tile_form=None, cache=False):
-
+                 desired_tile_form=None, cache=False,
+                 apikey=None):
+        if apikey is None:
+            raise ValueError("An apikey is now required to use Stamen maps. "
+                             "Visit https://maps.stadiamaps.com/ for more details.")
         # preset layer configuration
         layer_config = {
-          'terrain':            {'extension': 'png', 'opaque': True},
-          'terrain-background': {'extension': 'png', 'opaque': True},
-          'terrain-labels':     {'extension': 'png', 'opaque': False},
-          'terrain-lines':      {'extension': 'png', 'opaque': False},
-          'toner-background':   {'extension': 'png', 'opaque': True},
-          'toner':              {'extension': 'png', 'opaque': True},
-          'toner-hybrid':       {'extension': 'png', 'opaque': False},
-          'toner-labels':       {'extension': 'png', 'opaque': False},
-          'toner-lines':        {'extension': 'png', 'opaque': False},
-          'toner-lite':         {'extension': 'png', 'opaque': True},
-          'watercolor':         {'extension': 'jpg', 'opaque': True},
+          'stamen_terrain':            {'extension': 'png', 'opaque': True},
+          'stamen_terrain_background': {'extension': 'png', 'opaque': True},
+          'stamen_terrain_labels':     {'extension': 'png', 'opaque': False},
+          'stamen_terrain_lines':      {'extension': 'png', 'opaque': False},
+          'stamen_toner_background':   {'extension': 'png', 'opaque': True},
+          'stamen_toner':              {'extension': 'png', 'opaque': True},
+          'stamen_toner_hybrid':       {'extension': 'png', 'opaque': False},
```

Comment:
Toner hybrid is now deprecated according to https://docs.stadiamaps.com/map-styles/stamen-toner/